src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseExportService.java

```diff
@@ -149,6 +158,131 @@ private void deleteTempZipRepoFiles(List<Path> pathsToZipeedRepos) {
         }
     }
 
+    /**
+     * downloads all repos of the exercise and runs JPlag
+     *
+     * @param programmingExerciseId the id of the programming exercises which should be checked
+     * @return a zip file that can be returned to the client
+     * @throws ExitException is thrown if JPlag exits unexpectedly
+     * @throws IOException is thrown for file handling errors
+     */
+    public File checkPlagiarism(long programmingExerciseId) throws ExitException, IOException {
+        // TODO: offer the following options in the client
+        // 1) filter empty submissions, i.e. repositories with no student commits
+        // 2) filter submissions with a result score of 0%
+
+        ProgrammingExercise programmingExercise = programmingExerciseRepository.findWithAllParticipationsById(programmingExerciseId).get();
+
+        downloadRepositories(programmingExercise);
+
+        var output = "output";
+        var projectKey = programmingExercise.getProjectKey();
+        var outputFolder = REPO_DOWNLOAD_CLONE_PATH + (REPO_DOWNLOAD_CLONE_PATH.endsWith(File.separator) ? "" : File.separator) + projectKey + "-" + output;
+
+        File outputFolderFile = new File(outputFolder);
+        outputFolderFile.mkdirs();
+
+        String programmingLanguage = getJPlagProgrammingLanguage(programmingExercise);
+
+        var repoFolder = REPO_DOWNLOAD_CLONE_PATH + (REPO_DOWNLOAD_CLONE_PATH.endsWith(File.separator) ? "" : File.separator) + projectKey;
+        var templateRepoName = versionControlService.get().getRepositorySlugFromUrl(programmingExercise.getTemplateParticipation().getRepositoryUrlAsUrl());
+        String[] args = new String[] { "-l", programmingLanguage, "-r", outputFolder, "-s", repoFolder, "-bc", templateRepoName, "-vq" };
```

Comment:
Codacy found an issue: [Local variable 'args' could be declared final](https://app.codacy.com/gh/ls1intum/Artemis/pullRequest?prid=5968852)